### ISSUE TITLE
Needed basic widgets

### ISSUE THREAD

**Issue by ozgurakkurt** [xi-editor/druid#5]
make a list of widgets to be implemented

**Comment by Engid** [comment]
Knob and slider

**Comment by futurepaul** [comment]
Is a scrollable view the sort of thing that would be a widget? That's a high priority feature for me, and I'd like to help on it if possible, but I don't know exactly where in the stack that's going to be implemented.

**Comment by raphlinus** [comment]
Yes, it's in scope. However, scrolling is *invariably* one of the more complex architectural pieces of UI toolkit design. Does it try to instantiate the child widgets lazily? Asynchronously? Part of it depends on the expected scale. I need to do some design work.

This will in fact be a major priority soon in the font editor, because of the scrolling glyph list (on the order of 60k glyphs for a CJK font).

**Comment by futurepaul** [comment]
I'm not sure if this correlates to how druid is designed, but I think a big part of the React Native rearchitecture is [about making scrolling more synchronous](https://shockoe.com/blog/the-future-of-react-native-with-fabric/).

Anyway, look forward to your design work!

**Comment by raphlinus** [comment]
Rearchitecture seems to go with the territory, Flutter rearchitected their scrolling a year or so ago. It's [pretty complicated](https://medium.com/flutter-io/slivers-demystified-6ff68ab0296f) under the hood. I'm personally leaning towards doing the simplest thing (eagerly instantiating the children) and then figuring out where that falls short.

**Comment by futurepaul** [comment]
Bumping this with a progress report.

With caveats, here's what is functional right now:
- [x] Align
- [x] Button
- [x] Checkbox
- [x] Flex
- [x] Label (single line)
- [x] Padding
- [x] Progress Bar
- [x] Scroll
- [x] SizedBox
- [x] Slider
- [x] Textbox (single line)

Here's what's WIP afaik:
- [ ] Container (@zaynetro)
- [ ] List (@raphlinus)

Other basic widgets / extended functionality I believe we need:
- [ ] Combo Box
- [ ] Dropdown List
- [ ] Dropdown Menu
- [ ] Multiline Label
- [ ] Multiline Textbox
- [ ] Radio Button

What else do we need for a minimum widget set? I'm especially interested in anything Runebender-motivated.

**Comment by futurepaul** [comment]
I've lapsed in keeping this updated, we actually have [a toggle widget](https://github.com/xi-editor/druid/blob/master/druid/src/widget/switch.rs) now!

**Comment by jlgerber** [comment]
I would like to suggest two more advanced widgets - a table and a tree. 
(eg GtkTable and GtkTreeView)
These widgets are both pretty ubiquitous in non-trivial real world applications.

**Comment by rjwittams** [comment]
I've got a table in progress at.  https://github.com/rjwittams/druid_table
